packages/cysync-core/src/hooks/useOnboardingCheckpoint.ts

```diff
@@ -0,0 +1,12 @@
+import { useEffect } from 'react';
+
+import { keyValueStore } from '~/utils';
+
+export const useOnboardingCheckpoint = (path: string) => {
+  const updateCheckpoint = async () => {
+    await keyValueStore.checkpointPath.set(path);
+  };
+  useEffect(() => {
+    updateCheckpoint();
+  }, []);
```

Comment:
Rather than taking the location as props, can we get the location directly from react-router?